packages/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart

```diff
@@ -340,17 +338,29 @@ class CommitLogKeyStore
   /// Called in init method of commitLog to initialize on server start-up.
   Future<Map<String, CommitEntry>> _getCommitIdMap() async {
     var keyMap = <String, CommitEntry>{};
-    var values = await _getValues();
-    for (var value in values) {
+    Iterable iterable = (_getBox() as Box).values;
+    for (var value in iterable) {
       if (value.commitId == null) {
         _logger.finest(
             'CommitID is null for ${value.atKey}. Skipping to update entry into commitLogCacheMap');
         continue;
       }
-      // If keyMap contains the key, update the commitId in the map with greater commitId.
+      // The reason we remove and add is that, the map which is a LinkedHashMap
+      // should have data in the following format:
+      // {
+      //  {k1, v1},
+      //  {k2, v2},
+      //  {k3, v3}
+      // }
+      // such that v1 < v2 < v3
+      //
+      // If a key exist in the _commitLogCacheMap, updating the commit entry will
+      // overwrite the existing key resulting into an unsorted map.
+      // Hence remove the key and insert at the last ensure the entry with highest commitEntry
+      // is always at the end of the map.
```

Comment:
Thanks for the comment explaining the reasoning here! Will help future maintainers.